test/src/test/java/org/corfudb/integration/CorfuStoreIT.java

```diff
@@ -480,6 +488,115 @@ public void alterTableUsingDropTest() throws Exception {
+                .ContactBookId.class.getMethod("getDefaultInstance").invoke(null);
+        ManagedResources defaultMetadataMessage = (ManagedResources) ManagedResources
+                .class.getMethod("getDefaultInstance").invoke(null);
+
+        Table<Uuid, ExampleSchemas.ContactBookId, ManagedResources> badTable = new Table<Uuid, ExampleSchemas.ContactBookId, ManagedResources>(
+                TableParameters.<Uuid, ExampleSchemas.ContactBookId, ManagedResources>builder()
+                        .namespace(someNamespace)
+                        .fullyQualifiedTableName(getFullyQualifiedTableName(someNamespace, tableName))
+                        .kClass(Uuid.class)
+                        .vClass(ExampleSchemas.ContactBookId.class)
+                        .mClass(ManagedResources.class)
+                        .valueSchema(defaultValueMessage)
+                        .metadataSchema(defaultMetadataMessage)
+                        .schemaOptions(CorfuOptions.SchemaOptions.getDefaultInstance())
+                        .secondaryIndexesDisabled(true)
+                        .build(),
+                runtime,
+                runtime.getSerializers().getSerializer(ProtobufSerializer.PROTOBUF_SERIALIZER_CODE),
+                null,
+                new HashSet<>(Collections.singletonList(LOG_REPLICATOR_STREAM_INFO.getStreamId())));
+
+        // Now this badTable is completely hidden from both the TableRegistry and the Serializer!
+        TxnContext tx = corfuStore.txn(someNamespace);
+        long timestamp = System.currentTimeMillis();
+        ExampleSchemas.ContactBookId badSchema = ExampleSchemas.ContactBookId.newBuilder()
+                .setName(tableName).build();
+        tx.putRecord(badTable, key, badSchema,
+                ManagedResources.newBuilder()
+                        .setCreateTimestamp(timestamp).build());
+        tx.commit();
+
+        runtime.shutdown();
+
+        runtime = createRuntime(singleNodeEndpoint);
+        corfuStore = new CorfuStore(runtime);
+
+        // Re-open the table with the same schema and test if it hits a serializer exception
+        // as it would when a new code is trying to open something from an older instance
+        Table<Uuid, SampleSchema.EventInfo, ManagedResources> tableV2 = corfuStore.openTable(
+                someNamespace,
+                tableName,
+                Uuid.class,
+                SampleSchema.EventInfo.class,
+                ManagedResources.class,
+                // TableOptions includes option to choose - Memory/Disk based corfu table.
+                TableOptions.builder().build());
+
+        tx = corfuStore.txn(someNamespace);
+        boolean didAssertFire = false;
+        try {
+            tx.clear(tableV2);
+        } catch (TransactionAbortedException e) {
+            assertThat(e.getAbortCause()).isEqualTo(AbortCause.UNDEFINED);
+            didAssertFire = true;
+        }
+        assert(didAssertFire);
```

Comment:
Can we use the assertj or junit assert instead because these assertions are disabled by default?
Ref: https://www.baeldung.com/java-assert

Reply:
+1, I think we can use something like `Assert.fail("Expected TransactionAbortedException was not thrown")` after the clear as an alternative.

Reply:
done, thanks